lib/queuePopulator/KafkaLogConsumer/ListRecordStream.js

```diff
@@ -0,0 +1,82 @@
+const stream = require('stream');
+
+class ListRecordStream extends stream.Transform {
+    /**
+     * @constructor
+     * @param {Logger} logger logger
+     */
+    constructor(logger) {
+        super({ objectMode: true });
+        this._logger = logger;
+    }
+
+    /**
+     * Maps change stream operation type to
+     * backbeat supported types
+     * @param {string} operationType mongo opetation type
+     * @returns {string} supported operation type
+     */
+    _getType(operationType) {
+        switch (operationType) {
+            case 'insert':
+            case 'update':
+            case 'replace':
+                return 'put';
+            case 'delete':
+                return 'delete';
+            default:
+                this._logger.warn('Got unsupported operation', {
+                    method: 'ListRecordStream._getType',
+                    operationType,
+                });
+                return 'none';
+        }
+    }
+
+    /**
+     * parses json strings
+     * @param {string} blob sringified json
+     * @returns {Object} parsed json
+     */
+    _parseJson(blob) {
+        try {
+            let parsed = JSON.parse(blob);
+            if (typeof parsed === 'string') {
+                parsed = this._parseJson(parsed);
+            }
+            return parsed;
+        } catch (err) {
+            this._logger.warn('Got invalid kafka message value field format', {
+                method: 'ListRecordStream._parseJson',
+                value: blob,
+            });
+            return {};
+        }
+     }
+
+    /**
+     * Formats change stream entries
+     * @param {Object} data chunk of data
```

Comment:
Would really prefer a typedef for objects like this, I simply have no idea what is actually in data unless I analyze line by line if you use all parts of it.